mysql-test/suite/innodb/t/foreign_null.test

```diff
@@ -0,0 +1,110 @@
+--source include/have_innodb.inc
+let $algorithm=`select regexp_replace('$MTR_COMBINATIONS', 'innodb,\|,innodb', '')`;
+let $error_code = 0;
+
+if ($algorithm == "COPY") {
+ let $error_code = ER_FK_COLUMN_CANNOT_CHANGE;
+}
+
+if ($algorithm == "INPLACE") {
+ let $error_code = ER_FK_COLUMN_NOT_NULL;
+}
+
+# MODIFY CHILD COLUMN..NOT NULL ON UPDATE CASCADE.. PARENT COLUMN NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, FOREIGN KEY(f1) REFERENCES t1(f2) ON UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL,ALGORITHM=$algorithm;
+DROP TABLE t2, t1;
+
+# MODIFY CHILD COLUMN..NOT NULL ON DELETE|ON UPDATE SET NULL
+CREATE TABLE t1(f1 INT, f2 INT, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT, FOREIGN KEY(f1) REFERENCES t1(f1) ON UPDATE SET NULL, FOREIGN KEY (f2) REFERENCES t1(f2) ON DELETE SET NULL)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t2 MODIFY COLUMN f2 INT NOT NULL,ALGORITHM=$algorithm;
+
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t2 MODIFY COLUMN f1 INT NOT NULL,ALGORITHM=$algorithm;
+DROP TABLE t2, t1;
+
+
+let $error_code = ER_FK_COLUMN_CANNOT_CHANGE_CHILD;
+
+# MODIFY PARENT COLUMN.. NULL.. ON UPDATE CASCADE CHILD.. NOT NULL
+CREATE TABLE t1(f1 INT, f2 INT NOT NULL, PRIMARY KEY(f1), KEY(f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT NOT NULL, FOREIGN KEY(f1) REFERENCES t1(f2) ON UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t1 MODIFY COLUMN f2 INT,ALGORITHM=$algorithm;
+DROP TABLE t2, t1;
+
+# FOREIGN KEY CONSTRAINT FOR MULTIPLE COLUMNS
+CREATE TABLE t1(f1 INT NOT NULL, f2 INT NOT NULL,
+		INDEX(f1, f2))ENGINE=InnoDB;
+
+CREATE TABLE t2(f1 INT NOT NULL, f2 INT NOT NULL,
+		INDEX(f1, f2),
+		FOREIGN KEY(f1, f2) REFERENCES t1(f1, f2) ON
+		UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t1 MODIFY COLUMN f1 INT,ALGORITHM=$algorithm;
+DROP TABLE t2, t1;
+
+CREATE TABLE t1(f1 INT, f2 INT NOT NULL, INDEX(f1, f2))ENGINE=InnoDB;
+CREATE TABLE t2(f1 INT, f2 INT NOT NULL, INDEX(f1, f2),
+		FOREIGN KEY(f1, f2) REFERENCES t1(f1, f2) ON
+		UPDATE CASCADE)ENGINE=InnoDB;
+replace_result ,ALGORITHM=COPY '' ,ALGORITHM=INPLACE '';
+--error $error_code
+eval ALTER TABLE t1 MODIFY COLUMN f2 INT,ALGORITHM=$algorithm;
+DROP TABLE t2, t1;
+
+if ($algorithm == "COPY") {
+ let $error_code = ER_FK_COLUMN_CANNOT_CHANGE;
+}
+
+if ($algorithm == "INPLACE") {
+ let $error_code = ER_FK_COLUMN_NOT_NULL;
+}
+
+CREATE TABLE t1(f1 INT, f2 INT, INDEX(f1, f2))ENGINE=InnoDB;
```

Comment:
We are back to testing the child table CREATE/ALTER. Can you please add a comment for each test similar to the tests in the beginning.